cmd/telliot/main.go

```diff
@@ -132,280 +140,300 @@ func AddDBToCtx(remote bool) error {
-			addr.addr = commonAddress
+func (d dataserverCmd) Run(logger log.Logger) error {
+	// Create os kill sig listener.
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+
+	var ds *ops.DataServerOps
+	var err error
+	err = AddDBToCtx(true)
+	if err != nil {
+		return errors.Wrapf(err, "initializing database")
+	}
+	ch := make(chan os.Signal)
+	ds, err = ops.CreateDataServerOps(ctx, logger, ch)
+	if err != nil {
+		return errors.Wrapf(err, "creating data server")
+	}
+	// Start and wait for it to be ready
+	if err := ds.Start(ctx); err != nil {
+		return errors.Wrapf(err, "starting data server")
+	}
+	<-ds.Ready()
+
+	// Wait for kill sig.
+	<-c
+	// Notify exit channels.
+	ch <- os.Interrupt
+
+	cnt := 0
+	start := time.Now()
+	for {
+		cnt++
+		dsStopped := false
+
+		if ds != nil {
+			dsStopped = !ds.Running
+		} else {
+			dsStopped = true
 		}
-		ExitOnError(ops.Balance(ctx, clt, cont.Getter, addr.addr), "checking balance")
+
+		if !dsStopped && cnt > 60 {
+			level.Warn(logger).Log("msg", "taking longer than expected to stop operations", "waited", time.Since(start))
+		} else if dsStopped {
+			break
+		}
+		time.Sleep(500 * time.Millisecond)
 	}
+	level.Info(logger).Log("msg", "main shutdown complete")
+	return nil
+}
 
+func (l logLevel) AfterApply(ctx *kong.Context) error {
+	logger := util.GetLogger(string(l))
+	ctx.Bind(logger)
+	return nil
 }
 
-func disputeCmd(loggerSetup func(string) log.Logger, logLevel *string) func(*cli.Cmd) {
-	return func(cmd *cli.Cmd) {
-		cmd.Command("vote", "vote on an active dispute", voteCmd)
-		cmd.Command("new", "start a new dispute", newDisputeCmd)
-		cmd.Command("show", "show existing disputes", simpleCmd(ops.List, loggerSetup, logLevel))
+func (c configPath) AfterApply(ctx *kong.Context) error {
+	fmt.Println("config path")
+	err := config.ParseConfig(string(c))
+	if err != nil {
+		return errors.Wrapf(err, "parsing config")
 	}
+	client, contract, account, err := setup()
+	if err != nil {
+		return errors.Wrapf(err, "setting up variables")
+	}
+	ctx.BindTo(client, (*rpc.ETHClient)(nil))
+	ctx.Bind(contract)
+	ctx.Bind(account)
+
+	fmt.Println(contract.Address)
+
+	return nil
 }
 
-func voteCmd(cmd *cli.Cmd) {
-	disputeID := EthereumInt{}
-	cmd.VarArg("DISPUTE_ID", &disputeID, "dispute id")
-	supports := cmd.BoolArg("SUPPORT", false, "do you support the dispute? (true|false)")
-	cmd.Action = func() {
-		ExitOnError(ops.Vote(ctx, clt, cont, acc, disputeID.Int, *supports), "vote")
+func (c logConfigPath) AfterApply(ctx *kong.Context) error {
+	err := util.ParseLoggingConfig(string(c))
+	if err != nil {
+		return errors.Wrapf(err, "parsing log config")
 	}
+	return nil
 }
 
-func newDisputeCmd(cmd *cli.Cmd) {
+func (n newDisputeCmd) Run(logger log.Logger, client rpc.ETHClient, contract tellorCommon.Contract, account tellorCommon.Account) error {
 	requestID := EthereumInt{}
+	err := requestID.Set(n.requestID)
+	if err != nil {
+		return errors.Wrapf(err, "parsing argument")
```

Comment:
It seems that instead of this kong has validation so all these checks can happen in the `Validate` function
https://github.com/alecthomas/kong#validation

`func (n newDisputeCmd) Validate(....`